sqlglot/parser.py

```diff
@@ -1350,8 +1350,24 @@ def _parse_with(self, skip_with_token: bool = False) -> t.Optional[exp.Expressio
 
     def _parse_cte(self) -> exp.Expression:
         alias = self._parse_table_alias()
-        if not alias or not alias.this:
-            self.raise_error("Expected CTE to have alias")
+
+        # Alternative Clickhouse syntax: https://clickhouse.com/docs/en/sql-reference/statements/select/with/
+        if (
+            not alias
+            or not alias.this
+            or not self._match_pair(TokenType.ALIAS, TokenType.L_PAREN, advance=False)
+        ):
+            statement: t.Optional[exp.Expression]
+
+            if self._match(TokenType.ALIAS):
+                statement = self.expression(exp.Alias, this=alias, alias=self._parse_id_var())
+            else:
+                statement = self._parse_statement()
+
+            if not isinstance(statement, exp.Alias):
+                self.raise_error("Expected CTE to have alias")
+
+            return self.expression(exp.CTE, this=statement, alias=statement and statement.alias)
```

Comment:
the statement is no longer an alias,  i think probably for CTE, we should add a new field to represent this as an expression right? this is kind of a hack

Reply:
Well the syntax is always something like `<foo> AS <bar>`, so the idea was that:

- For anything we're already supporting, the behavior won't change.
- For clickhouse-specific CTEs, we'll parse the `<foo> AS <bar>` as an `exp.Alias` node and rely on that.